plugins/code-climate/src/api/production-api.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  CodeClimateData,
+  CodeClimateRepoData,
+  CodeClimateMaintainabilityData,
+  CodeClimateTestCoverageData,
+  CodeClimateIssuesData,
+} from './code-climate-data';
+import { CodeClimateApi } from './code-climate-api';
+import { DiscoveryApi, IdentityApi } from '@backstage/core-plugin-api';
+import { Duration } from 'luxon';
+import humanizeDuration from 'humanize-duration';
+
+const basicIssuesQuery = `${encodeURI('page[size]')}=1&${encodeURI(
+  'filter[status][$in][]',
+)}=open&${encodeURI('filter[status][$in][]')}=confirmed&${encodeURI(
+  'page[size]',
+)}=1&${encodeURI('filter[categories][$in][]')}=`;
+
+const codeSmellsQuery = `${basicIssuesQuery}Complexity`;
+
+const duplicationQuery = `${basicIssuesQuery}Duplication`;
+
+const otherIssuesQuery = `${basicIssuesQuery}Bug Risk`;
+
+export class ProductionCodeClimateApi implements CodeClimateApi {
+  constructor(
+    private readonly discoveryApi: DiscoveryApi,
+    private readonly identityApi?: IdentityApi,
+  ) {}
+
+  async fetchAllData(
+    apiUrl: string,
+    options: any,
+    repoID: string,
+    snapshotID: string,
+    testReportID: string,
+  ): Promise<any> {
+    const [
+      maintainabilityResponse,
+      testCoverageResponse,
+      codeSmellsResponse,
+      duplicationResponse,
+      otherIssuesResponse,
+    ] = await Promise.all([
+      await fetch(`${apiUrl}/repos/${repoID}/snapshots/${snapshotID}`, options),
+      await fetch(
+        `${apiUrl}/repos/${repoID}/test_reports/${testReportID}`,
+        options,
+      ),
+      await fetch(
+        `${apiUrl}/repos/${repoID}/snapshots/${snapshotID}/issues?${codeSmellsQuery}`,
+        options,
+      ),
+      await fetch(
+        `${apiUrl}/repos/${repoID}/snapshots/${snapshotID}/issues?${duplicationQuery}`,
+        options,
+      ),
+      await fetch(
+        `${apiUrl}/repos/${repoID}/snapshots/${snapshotID}/issues?${otherIssuesQuery}`,
+        options,
+      ),
+    ]);
+
+    if (
+      (maintainabilityResponse.status >= 400 &&
```

Comment:
Could all of these instead be `if (!xResponse.ok)`? I presume that even 3xx return codes won't be acceptable, so there's probably no need to check the actual code ranges like this